sklearn/ensemble/_hist_gradient_boosting/grower.py

```diff
@@ -430,18 +452,10 @@ def _initialize_root(self, gradients, hessians):
 
         if self.interaction_cst is not None:
             self.root.interaction_cst_indices = range(len(self.interaction_cst))
-            allowed_features = set().union(*self.interaction_cst)
-            self.root.allowed_features = np.fromiter(
-                allowed_features, dtype=np.uint32, count=len(allowed_features)
-            )
+            self.root.allowed_features = allowed_features
 
-        tic = time()
-        self.root.histograms = self.histogram_builder.compute_histograms_brute(
-            self.root.sample_indices, self.root.allowed_features
-        )
-        self.total_compute_hist_time += time() - tic
+        self.root.histograms = histograms
 
-        tic = time()
         self._compute_best_split_and_push(self.root)
         self.total_find_split_time += time() - tic
```

Comment:
It seems previously we were just timing the _compute_best_split_and_push function but now we are including all the code that involves the root construction as well. Is this intended?

Reply:
I thought about that and like it better now because everything is accounted for in timing.

Reply:
Then shouldn't we rename this variable to something like `total_construct_root_and_find_split_time`?

Reply:
No. I think it’s fine as it is, even better than before.